src/Format/Arguments.php

```diff
@@ -61,17 +62,43 @@ public function useArgument($name, $neededType)
         return $id;
     }
 
-    public function getPhp()
+    public function useSubtractiveArgument($nameA, $nameB, $neededTypeA, $neededTypeB)
     {
-        $statements = array_map('self::getInputStatement', array_flip($this->output));
-        $array = self::getArray($statements);
-        return self::getAssignment('$output', $array);
+        if (!array_key_exists($nameA, $this->input) || !array_key_exists($nameB, $this->input)) {
+            return null;
+        }
+
+        $userTypeA = gettype($this->input[$nameA]);
+        $userTypeB = gettype($this->input[$nameB]);
+
+        if (($userTypeA !== 'NULL') && ($userTypeA !== $neededTypeA)) {
+            return null;
+        }
+
+        if (($userTypeB !== 'NULL') && ($userTypeB !== $neededTypeB)) {
+            return null;
+        }
+
+        $safeNameA = var_export($nameA, true);
+        $idA = &$this->output["\$input[{$safeNameA}]"];
```

Comment:
Would it make sense to abstract away the details of constructing a "\$input[{$safeNameA}]" from the parameter name?